dango/dex/src/core/order_filling.rs

```diff
@@ -98,14 +98,7 @@ fn fill_bids(
         // For quote, in case the order is filled at a price better than the
         // limit price, refund the unused deposit.
         let refund_base = filled_base.checked_sub(fee_base)?;
-        let mut refund_quote = filled_base.checked_mul(order_price - clearing_price)?;
-
-        // For market orders, refund the remaining (unfilled) amount, as market
-        // orders are immediate-or-cancel.
-        if let Order::Market(market_order) = order {
-            let remaining_in_quote = market_order.remaining.checked_mul(market_order.price)?;
-            refund_quote.checked_add_assign(remaining_in_quote)?;
-        }
+        let refund_quote = filled_base.checked_mul(order_price - clearing_price)?;
```

Comment:
This logic is moved to `cron.rs` (line 603). It is related to canceling the market order, not fulfilling it, so it's inappropriate that it's placed here, in `fill_bids`.